src/main/java/com/oltpbenchmark/benchmarks/templated/util/TemplatedValue.java

```diff
@@ -0,0 +1,184 @@
+package com.oltpbenchmark.benchmarks.templated.util;
+
+import com.oltpbenchmark.distributions.ScrambledZipfianGenerator;
+import com.oltpbenchmark.distributions.ZipfianGenerator;
+import com.oltpbenchmark.util.JDBCSupportedType;
+import com.oltpbenchmark.util.TextGenerator;
+import java.util.Random;
+
+/**
+ * This class is used to store information about the values used in templated benchmarks. It can
+ * hold static values but also generators for value distributions
+ */
+public class TemplatedValue {
+  ValueGenerator distribution;
+  Long min;
+  Long max;
+  Long seed;
+  String value;
+  String minS;
+  String maxS;
+  Object generatorObject;
+
+  /**
+   * @param value Value that is used if no distribution is given
+   */
+  public TemplatedValue(String value) {
+    if (value.equals("null") || value.length() < 1) {
+      this.value = null;
+    } else {
+      this.value = value;
+    }
+  }
+
+  /**
+   * @param distribution The desired value distribution
+   * @param min Minimum value. Default is 0
+   * @param max Maximum value. Default is min + 1
+   * @param seed The seed for the random generator. Default is 0
+   * @param value Value that is used if no distribution is given
+   */
+  public TemplatedValue(String distribution, String min, String max, String seed) {
```

Comment:
Re datetime parsing, couple options I can see here:
- Make different generic types for TemplateValue (probably extra work)
- Pass the "param_type" in with the constructor and use it do the extra min/max parsing logic, as well as some sanity check asserts for the various `getNext...` calls (e.g., get next string, for a non-string type is probably a mistake).

Reply:
I added the param type to the TemplatedValue and changed the parse logic now